selenium/firefox/apt/Dockerfile.tmpl

```diff
@@ -6,5 +6,18 @@ ARG CLEANUP
 RUN  \
         ( [ "$CLEANUP" != "true" ] && rm -f /etc/apt/apt.conf.d/docker-clean ) || true && \
         apt-get update && \
-        apt-get -y --no-install-recommends install firefox=$VERSION && \
+        apt-get -y --no-install-recommends install pulseaudio firefox=$VERSION && \
         ($CLEANUP && rm -Rf /tmp/* && rm -Rf /var/lib/apt/lists/*) || true
+
+RUN useradd seluser \
+         --shell /bin/bash  \
+         --create-home \
+  && usermod -a -G sudo seluser \
+  && echo 'ALL ALL = (ALL) NOPASSWD: ALL' >> /etc/sudoers \
+  && echo 'seluser:secret' | chpasswd
+
+RUN mkdir /video
```

Comment:
Why do we need this directory? Can't we obtain it using volumes?

Reply:
When the Firefox container starts, it creates this folder (`/video`), and since now it is a non-root user, it will not have the proper grant. The error I get when this folder does not exist is the following:

    --- x11vnc loop: 1 ---

    2017/12/27 12:13:42 Loading configuration files...
    2017/12/27 12:13:42 Loaded configuration from [/etc/selenoid/browsers.json]
    2017/12/27 12:13:42 Using default containers log configuration because of: read error: open config/container-logs.json: no such file or directory
    2017/12/27 12:13:42 Failed to create video output dir /video: mkdir /video: permission denied

Reply:
Euh... Video output directory is expected to be created on host machine and then mounted as container directory via volumes (Selenoid does this automatically). So I think this change is useless.

Reply:
I've been looking into this a bit more. It seems selenoid is creating the folder `/video` inside the container. The source code is [here](https://github.com/aerokube/selenoid/blob/master/main.go). Since the user is not root anynore, this command makes the Firefox container to fail at startup. For that reason I need to create that folder owned by `selenium` user beforehand.